experiments/tf_trainer/common/text_preprocessor.py

```diff
@@ -193,12 +236,19 @@ def _get_word_idx_and_embeddings(embeddings_path: str,
     """
     word_to_idx = {}
     word_embeddings = []
-    with tf.gfile.Open(embeddings_path) as f:
+    if is_binary_embedding:
+      read_mode = 'rb'
+    else:
+      read_mode = 'r'
+    with tf.gfile.Open(embeddings_path, read_mode) as f:
       for idx, line in enumerate(f):
         if max_words and idx >= max_words:
           break
 
         values = line.split()
+        # Remove header when necessary.
+        if len(values) == 2 and idx == 0:
```

Comment:
Is it better to have this be automated or as a flag? My guess is that it's better to have an explicit flag than this check which could fail implicitly for lots of reasons.

Reply:
Hmm good point.
So the solution should be:  flag "skip_first_line"?

Will add it to next commit.